### PR TITLE
retry on all thrown errors by default

### LINKED ISSUE #88: retry on all thrown fetch errors
We used to retry on all thrown http errors: https://github.com/adobe/node-fetch-retry/blob/4161e438c4014f69a31f575578714764c17fdfee/index.js#L29

Then when we refactored the code, the behavior changed and only retried on `error.code` 5xx which brought in this regression: https://github.com/adobe/node-fetch-retry/issues/60
To be more thorough, we added a customizable function called `retryOnHttpError` and changed the default to retry on all FetchErrors: https://github.com/adobe/node-fetch-retry/pull/63

The goal now is to expand the behavior (in addition to this change: https://github.com/adobe/node-fetch-retry/issues/87) to call `retryOnHttpError` for all thrown errors, not just `FetchError`. This will make it easier to implement this issue: https://github.com/adobe/node-fetch-retry/issues/86

Developers can still use the custom `retryOnHttpError` function to add special behavior, like only retrying on `FetchError`'s.

### PR DESCRIPTION
**Pull request by jdelbick**
Fixes #88

## Description

Requires https://github.com/adobe/node-fetch-retry/issues/87 as a pre-requisite

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -296,6 +296,53 @@ describe('test `retryInit` function', () => {
...
+                this.name = 'AbortError';
+            }
+        }
+        assert.strictEqual(await retryOptions.retryOnHttpError(new FetchError('hello!', 'non-system')), true);
+        assert.strictEqual(await retryOptions.retryOnHttpError(new FetchError('hello!', 'system')), true);
+        assert.strictEqual(await retryOptions.retryOnHttpError(new FetchError('ECONNRESET', 'system')), true);
+        assert.strictEqual(await retryOptions.retryOnHttpError(new Error('ECONNRESET', 'system')), true);
+        assert.strictEqual(await retryOptions.retryOnHttpError(new AbortError('error!')), true);
+        // will be fixed to not retry on these by https://github.com/adobe/node-fetch-retry/issues/87
+        assert.strictEqual(retryOptions.retryOnHttpError(new TypeError('x is not a valid JSON')), true);
```
Comment (jdelbick) [inline test/index.test.js:344]: TODO: will be `false` once https://github.com/adobe/node-fetch-retry/issues/87 is done

**Review by alexkli — APPROVED** [review]
@jdelbick I changed this PR to go into the new `version3` branch.

**Comment by jdelbick** [comment]
@alexkli should I wait to merge this until https://github.com/adobe/node-fetch-retry/issues/87 is fixed & merged to v3?

File: index.js [hunk 2]
```diff
@@ -150,18 +150,8 @@ function checkParameters(retryOptions) {
...
-    if (error.name === 'FetchError' && error.type === 'system') {
-        console.error(`FetchError failed with code: ${error.code}; message: ${error.message}`);
-        return true;
-    } else if (error.name === 'AbortError') {
-        console.error(`AbortError failed with type: ${error.type}; message: ${error.message}`);
-        return true;
-    }
-    return false;
+function shouldRetryOnHttpError() {
+    return true;
```
Comment (alexkli) [inline index.js]: If this is always true, we could change its only usage on line ~102 to just


```javascript
    retryOnHttpError: ((typeof retryOptions.retryOnHttpError === 'function') && retryOptions.retryOnHttpError) ||
                ((error) => true),
```

**Review by alexkli — APPROVED** [review]
I think this can be merged already. See my additional inline comment.

**Pull request merged**